src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -26,7 +23,27 @@ internal sealed class NoneOperation : Operation
             OperationConstantValue = constantValue;
         }
 
-        public override IEnumerable<IOperation> Children { get; }
+        internal ImmutableArray<IOperation> Children { get; }
+
+        protected override IOperation GetCurrent(int slot, int index)
+            => slot switch
+            {
+                0 when index < Children.Length
+                    => Children[index],
+                _ => throw ExceptionUtilities.UnexpectedValue((slot, index))
+            };
+
+        protected override (bool hasNext, int nextSlot, int nextIndex) MoveNext(int previousSlot, int previousIndex)
+            => previousSlot switch
+            {
+                -1 when !Children.IsEmpty
+                    => (true, 0, 0),
+                0 when previousIndex + 1 < Children.Length
+                    => (true, 0, previousIndex + 1),
+                0 => (false, int.MinValue, int.MinValue),
+                _ => throw ExceptionUtilities.UnexpectedValue((previousSlot, previousIndex))
```

Comment:
It looks like manual implementations have the same issue around not fulfilling the contact.

Reply:
Yep, also corrected locally.

---
In reply to: [541186396](https://github.com/dotnet/roslyn/pull/49804#discussion_r541186396) [](ancestors = 541186396)